src/Common/TaskQueue/TaskQueue.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Temporal\Common\TaskQueue;
+
+use Temporal\Internal\Marshaller\Meta\Marshal;
+use Temporal\Internal\Traits\CloneWith;
+
+/**
+ * @link https://docs.temporal.io/docs/concepts/task-queues/
+ * @see \Temporal\Api\Taskqueue\V1\TaskQueue
+ */
+final class TaskQueue
```

Comment:
I don't think you need this at all. Users never specify `kind` and therefore anywhere that needs to accept a task queue can simply accept a string

Reply:
I've removed the `TaskQueue::withKind()` method, leaving only `TaskQueue::withName()`. It will be difficult to get rid of this DTO because of the automatic mapping. However, in methods that take `TaskQueue` from the user, it's possible to specify a `string`, not only a `TaskQueue` object: 

```php
// class StartWorkflowAction
public function withTaskQueue(string|TaskQueue $taskQueue): self
```